rust/private/rustc.bzl

```diff
@@ -178,10 +189,43 @@ def rustc_compile_action(
     if ctx.file.out_dir_tar:
         compile_inputs.append(ctx.file.out_dir_tar)
 
+    rpaths = _compute_rpaths(toolchain, output_dir, depinfo)
+
+    if not hasattr(ctx.fragments.cpp, "linkopts"):
+        fail("Bazel is too old, please use at least Bazel 0.18.0")
+    elif versions.is_at_least("0.18.0", BAZEL_VERSION):
+        user_link_flags = ctx.fragments.cpp.linkopts
+    else:
+        user_link_flags = depset(ctx.fragments.cpp.linkopts)
+
     # Paths to cc (for linker) and ar
     cpp_fragment = ctx.host_fragments.cpp
-    cc = cpp_fragment.compiler_executable
-    ar = cpp_fragment.ar_executable
+    cc_toolchain = find_cpp_toolchain(ctx)
+    feature_configuration = cc_common.configure_features(
+        cc_toolchain = cc_toolchain,
+        requested_features = ctx.features,
+        unsupported_features = ctx.disabled_features,
+    )
+    ld = cc_common.get_tool_for_action(
```

Comment:
Tangentially related to this PR:

Its not clear to me how the linker gets selected. Is the target platform the same as the one specified (indirectly) in `find_cpp_toolchain`?

Reply:
C++ rules don't currently use platforms, they have their own thing controlled by `--crosstool_top`, `--cpu`, and `--compiler` (and their corresponding `host` variants, such as `--host_crosstool_top`. Here in the rule we are getting target C++ toolchain, when we're in the target configuration, and host toolchain when in host configuration (so if somebody just tells Bazel to `build //foo`, we build `foo` in target configuration. If somebody writes a genrule that has `foo` as a tool, Bazel will build it in host configuration). Did I answer your question?